back/app/services/side_fx_custom_field_option_service.rb

```diff
@@ -20,6 +20,12 @@ def before_destroy(custom_field_option, _current_user)
   end
 
   def after_destroy(frozen_custom_field_option, current_user)
+    custom_field = frozen_custom_field_option.custom_field
+    if custom_field&.resource_type == 'CustomForm' && custom_field&.input_type == 'select'
+      FormLogicService.new([frozen_custom_field_option.custom_field])
+        .remove_select_logic_option_from_custom_fields(frozen_custom_field_option)
+    end
```

Comment:
I'm not sure this is the best approach, as this won't result in a `LogActivityJob` to create an `Activity` for the resulting CustomField update.

Reply:
I think that's fine, as the update is a side effect (e.g. removing relationships when deleting something will not create an update activity either)